src/components/user-page/UserSettings.js

```diff
@@ -0,0 +1,119 @@
+import React, {Component} from 'react';
+import Header from "../header";
+import ImageUpload from "./ImageUpload";
+import axios from 'axios';
+import {ACCESS_TOKEN} from "../../constants/constants";
+import {getJwt} from "../../helpers/jwt";
+
+class UserSettings extends Component {
+    constructor(props) {
+        super(props);
+
+        this.getUser = this.getUser.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+
+        this.state = {
+            firstName: '',
+            lastName: '',
+            login: '',
+            password: '',
+            description: '',
+            avatar: ''
+        };
+    }
+
+    componentDidMount() {
+        this.getUser();
+    }
+
+    handleChange(event) {
+        this.setState({[event.target.name]: event.target.value});
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+
+        const {login, password, firstName, lastName, email, description, avatar} = this.state;
+    }
+
+    getUser(){
+        let token = getJwt();
+        var url = 'http://localhost:8080/users/' + localStorage.getItem("login");
+        axios.get(url,{
+            headers:{
+                Authorization: token
+            }
+        }).then(res =>{
+            this.setState({login: res.data.login});
+            this.setState({firstName: res.data.firstName});
+            this.setState({lastName: res.data.lastName});
+            this.setState({description: res.data.description});
+            this.setState({avatar: res.data.avatar});
```

Comment:
wow, it's work correct?)
better use
  this.setState({
login: res.data.login ,
firstName: res.data.firstName,
lastName: res.data.lastName,
description: res.data.description,
avatar: res.data.avatar
});

Reply:
to late, already merged into brach mate-